raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -671,8 +671,8 @@ private void transitionToResigned(List<ReplicaKey> preferredSuccessors) {
         resetConnections();
     }
 
-    private void transitionToVoted(ReplicaKey candidateKey, int epoch) {
-        quorum.transitionToVoted(epoch, candidateKey);
+    private void transitionToUnattachedVoted(ReplicaKey candidateKey, int epoch) {
+        quorum.addVotedState(epoch, candidateKey);
         maybeFireLeaderChange();
         resetConnections();
```

Comment:
Okay but note that if kraft is transitioning from "unattached" to "unattached voted", all of the pending Fetch request will be abandoned and resent.

Reply:
hm, I think we can remove `maybeFireLeaderChange()` too